chicago/models.py

```diff
@@ -168,3 +175,103 @@ def most_recent_past_city_council_meeting(cls):
             )
         else:
             return None
+
+
+class ChicagoPerson(Person):
+    class Meta:
+        proxy = True
+        app_label = "chicago"
+
+    # this is a very expensive query - do not use on listing pages
+    @property
+    def full_attendance(self):
+        attendance = []
+        events = []
+
+        current_legislative_session = LegislativeSession.objects.get(
+            start_date__lt=timezone.now(), end_date__gt=timezone.now()
+        )
+
+        # fetch all events for the current legislative session for committees they're on
+        for membership in self.current_memberships.all():
+            events.extend(
+                ChicagoEvent.objects.filter(
+                    participants__organization=membership.organization,
+                    start_date__gte=membership.start_date,
+                )
+                .filter(start_date__gte=current_legislative_session.start_date)
+                .filter(participants__entity_type="person")
+                .distinct()
+                .prefetch_related("participants")
+            )
+
+        for e in sorted(events, key=attrgetter("start_time"), reverse=True):
+            attended = e.participants.filter(person_id=self.id).exists()
+            attendance.append(
+                {
+                    "event": e,
+                    "attended": attended,
+                }
+            )
+
+        return attendance
+
+    @property
+    def attendance_percent(self):
+        attendance = self.full_attendance
+        if len(attendance) > 0:
+            attended = [a for a in attendance if a["attended"]]
+            return "{:.0%}".format(len(attended) / len(attendance))
+        else:
+            return 0
+
+    @property
+    def legislation_count(self):
+        return ChicagoBill.objects.filter(
+            sponsorships__person=self, sponsorships__primary=True
+        ).count()
+
+    @property
+    def election_status(self):
+        for p in settings.ALDER_EXTRAS:
```

Comment:
any reason not to add this to the model? 

where are you deriving this data from, btw?

Reply:
I put it in the settings to have a resilient place to store this information. We can't add it to the person model without modifying `django-councilmatic`. `ChicagoPerson` is an abstract model and `ChicagoPersonStatistic` is for a different purpose

The election status is derived from news articles, namely this one: https://blockclubchicago.org/2022/09/27/city-hall-shake-up-why-aldermen-are-retiring-in-droves-and-how-it-could-change-chicago-politics/

Reply:
that seems reasonable. one thing you can do is stuff things in on the extras property on the person model. 

if you let me know where you are sourcing the caucus information, then I might add those to the scraper and stuff them into extras.